app/scripts/utils/polygon-url.ts

```diff
@@ -47,10 +60,32 @@ export function polygonUrlEncode(
 ) {
   return featureCollection.features
     .map((feature) => {
-      const points = feature.geometry.coordinates[0]
-        // Remove last coordinate since it is repeated.
-        .slice(0, -1);
-      return encode(points);
+      return encodePolygon(feature.geometry);
     })
     .join(';');
 }
+
+export function encodeAois(aois: AoIFeature[]): string {
+  const encoded = aois.reduce((acc, aoi) => {
+    const encodedGeom = encodePolygon(aoi.geometry);
+    return [...acc, encodedGeom, toAoIid(aoi.id), !!aoi.selected];
+  }, []);
+  return JSON.stringify(encoded);
+}
+
+export function decodeAois(aois: string): AoIFeature[] {
+  const decoded = JSON.parse(aois) as string[];
+  const features = decoded.reduce<AoIFeature[]>((acc, current, i) => {
+    if (i % 3 === 0) {
+      const decodedFeature = decodeFeature(current) as AoIFeature;
+      return [...acc, decodedFeature];
+    } else {
+      const lastFeature = acc[acc.length - 1];
+      const prop = i % 3 === 1 ? 'id' : 'selected';
+      const newFeature = { ...lastFeature, [prop]: current };
+      acc[acc.length - 1] = newFeature;
+      return acc;
+    }
+  }, []);
+  return features!;
```

Comment:
I like how you stored the data in the array, however I found this piece of code a bit tricky to parse.
A possible approach would be to chunk the array in groups of 3 and handle it that way.

Just a suggestion, not a blocker.
```tsx
import { chunk } from 'lodash';

export function decodeAois(aois: string): AoIFeature[] {
  const decoded = JSON.parse(aois) as string[];

  const features: AoIFeature[] = chunk(decoded, 3).map((data) => {
    const [polygon, id, selected] = data;
    const decodedFeature = decodeFeature(polygon) as AoIFeature;
    return { ...decodedFeature, id, selected };
  });

  return features;
}
```

Reply:
@danielfdsilva I had no idea `chunk` existed. You made my day ❤️